src/components/pages/Login.test.tsx

```diff
@@ -0,0 +1,110 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { QueryClient, QueryClientProvider } from 'react-query'
+import { BrowserRouter as Router } from 'react-router-dom'
+import Login from './Login'
+
+const mockLoginUser = jest.fn()
+
+jest.mock('../../hooks/useLoginUser', () => {
+  return () => {
+    return {
+      mutate: mockLoginUser,
+      loading: false,
+      error: 'There is an error in login',
+      success: true,
+      user: {
+        username: 'kitty',
+        token: 'tokenString',
+      },
+    }
+  }
+})
+
+describe('Login', () => {
+  jest.spyOn(Object.getPrototypeOf(window.localStorage), 'setItem')
+  Object.setPrototypeOf(window.localStorage.setItem, jest.fn())
+
+  test('renders login form without crashing', () => {
+    render(
+      <QueryClientProvider client={new QueryClient()}>
+        <Router>
+          <Login />
+        </Router>
+      </QueryClientProvider>,
```

Comment:
<QueryClientProvider client={new QueryClient()}>
  <Router>
     ....

There is repeative part used for every render. We can create something like TestProvider which contains the common part such as those QueryClientProvider and Router.

Reply:
Just pushed up the changes. Is this ok?